### ISSUE TITLE
[CMake] Configure fails at HEAD: runtime/components/CMakeLists.txt references missing/moved CMakeLists.txt for `constrained_decoding` and `preprocessor`

### ISSUE THREAD

**Issue by ianbmacdonald** [google-ai-edge/LiteRT-LM#2932]
## Summary

A standard CMake build (`cmake -B build .`, the documented non-Bazel path) of LiteRT-LM at **HEAD
(`d051de38`)** fails immediately at the **configure** step — before any compilation — because
`runtime/components/CMakeLists.txt` references two component directories whose `CMakeLists.txt` is
no longer at the referenced path. This is a regression since **v0.13.0**, which configures cleanly.
The **Bazel** build path is unaffected.

## Error

```
CMake Error at runtime/components/CMakeLists.txt:18 (add_subdirectory):
  add_subdirectory given source "constrained_decoding" which is not an existing directory.

CMake Error at runtime/components/CMakeLists.txt:20 (add_subdirectory):
  The source directory .../runtime/components/preprocessor does not contain a CMakeLists.txt file.
```

`runtime/components/CMakeLists.txt` lines 18–21 (unchanged between v0.13.0 and HEAD):

```cmake
add_subdirectory(constrained_decoding)
add_subdirectory(embedding_lookup)
add_subdirectory(preprocessor)
add_subdirectory(tool_use)
```

## What changed (per-tag, from the public tree)

`CMakeLists.txt` presence at each path (HTTP status on raw.githubusercontent):

| path | v0.11.0 | v0.13.0 | HEAD |
|---|:---:|:---:|:---:|
| `runtime/components/constrained_decoding/CMakeLists.txt` | ✅ 200 | ✅ 200 | ❌ 404 |
| `runtime/components/logits_processor/constrained_decoding/CMakeLists.txt` | ❌ 404 | ❌ 404 | ✅ 200 |
| `runtime/components/preprocessor/CMakeLists.txt` | ✅ 200 | ✅ 200 | ❌ 404 |

Two observations:

1. **`constrained_decoding`** — the `CMakeLists.txt` now lives under
   `runtime/components/logits_processor/constrained_decoding/`, but line 18's
   `add_subdirectory(constrained_decoding)` still points at the old flat path.
2. **`preprocessor`** — `runtime/components/preprocessor/` still ships its headers and Bazel
   `BUILD`, but its `CMakeLists.txt` is absent at HEAD (it was present through v0.13.0).

## Reproduction

```bash
git clone https://github.com/google-ai-edge/LiteRT-LM.git && cd LiteRT-LM
git checkout d051de38
cmake -B build -G "Unix Makefiles" .
# → configure fails at runtime/components/CMakeLists.txt:18 and :20
```

Reproduces on a plain **x86_64** host — no cross-compilation or special toolchain required.

## Workaround

Pinning to **`v0.13.0`** configures and builds via CMake, since both `CMakeLists.txt` files exist
at the paths lines 18/20 reference.

## Suspected fix (unverified)

Updating line 18 to `add_subdirectory(logits_processor/constrained_decoding)` and restoring
`runtime/components/preprocessor/CMakeLists.txt` would likely realign the CMake tree with its
current layout. The Bazel `BUILD` files for both components are present and build correctly at
HEAD, so this appears to be limited to the CMake build path.

## Environment

- LiteRT-LM `d051de38` (also observed on the current `main`)
- CMake 4.2 / 3.30; GCC 13–15; Ubuntu; x86_64 (arch-independent)

**Comment by nicholascordova01** [comment]
Notice regarding release-artifact integrity. For the awareness of maintainers, downstream consumers, and any reviewer relying on this repository's version lineage: the v0.14.0 release tag, originally created on commit a39354a (on or about July 1, 2026), was subsequently force-moved to commit 80f301f (on or about July 8, 2026), and release binaries were replaced in place under the same version number and URL, producing different checksums (see the independently reported issues on this tracker regarding the binary/source mismatch and the Swift package binary replacement). Because the version label no longer points to the commit originally published under it, reviewers examining "v0.14.0" should be aware that the current tagged code differs from the code originally published and deployed under that version. The undersigned requests that maintainers preserve the original commit a39354a, the original tag state, and the original pre-replacement binaries, and that release tags and artifacts not be further altered while the matters identified below are pending.

PENDING MATTERS (BY OFFICIAL IDENTIFIER). Department of Defense OIG complaint 202607070922061330_hotline_webform; Texas Attorney General TCP-050308, TCP-049257, and TCP-053761; FedRAMP Support Ticket 63271; FBI IC3 Submission 07ce662f05d941bb9a49535a2d708de9; Federal Trade Commission Report 202894092; Parker County, Texas Official Public Records Document 202618373; USPTO OED referral (ID 2-01092491). The repository owner has actual notice of these matters on its own issue tracker (Issue #2786).

**Comment by nireekshak** [comment]
Hi @ianbmacdonald,

Thanks for the detailed analysis and reproduction steps. We'll investigate the CMake configuration and verify the necessary changes to restore the documented CMake build path and get back to you.

**Comment by whhone** [comment]
Closing since this specific issue was fixed.
